src/main/java/org/mule/extension/ftp/internal/AbstractFtpCopyDelegate.java

```diff
@@ -55,19 +58,26 @@ public AbstractFtpCopyDelegate(FtpCommand command, FtpFileSystem fileSystem) {
   public void doCopy(FileConnectorConfig config, FileAttributes source, URI targetUri, boolean overwrite) {
     ConnectionHandler<FtpFileSystem> writerConnectionHandler;
     final FtpFileSystem writerConnection;
+    String path = source.getPath();
     try {
       writerConnectionHandler = getWriterConnection(config);
       writerConnection = writerConnectionHandler.getConnection();
     } catch (ConnectionException e) {
       throw command
           .exception(format("FTP Copy operations require the use of two FTP connections. An exception was found trying to obtain second connection to"
-              + "copy the path '%s' to '%s'", source.getPath(), targetUri.getPath()), e);
+              + "copy the path '%s' to '%s'", path, targetUri.getPath()), e);
     }
     try {
       if (source.isDirectory()) {
-        copyDirectory(config, createUri(source.getPath()), targetUri, overwrite, writerConnection);
+        copyDirectory(config, createUri(path), targetUri, overwrite, writerConnection);
+        if (LOGGER.isTraceEnabled()) {
+          LOGGER.trace("Copied directory {} to {}", path, targetUri);
```

Comment:
Shouldn't it be `targetUri.getPath()`? (Or do the same that you did with `String path = source.getPath(); `)